omniscidb/QueryEngine/JoinHashTable/PerfectJoinHashTable.cpp

```diff
@@ -971,11 +983,44 @@ llvm::Value* PerfectJoinHashTable::codegenSlot(const CompilationOptions& co,
   }
   const auto key_lvs = code_generator.codegen(key_col, true, co);
   CHECK_EQ(size_t(1), key_lvs.size());
+  auto key_col_type = key_col->type();
+
+  if (val_col->isVirtual()) {
+    // short-circuit hash table load. we know the key will be the row number we want to
+    // access, so just return the key
+    CHECK(val_col_var);  // make sure this virtual column is a column
+
+    std::vector<llvm::Value*> hash_join_idx_args{
+        executor_->cgen_state_->castToTypeIn(key_lvs.front(), 64)};
+
+    const auto& query_info = getInnerQueryInfo(val_col_var).info;
+    if (query_info.getPhysicalNumTuples() == 0) {
```

Comment:
Just curious why we can't always use `rhs_source_col_range_`. In theory, for an empty table, we should have some invalid ranges for all columns where max < min.

Reply:
Looks like it's set for `rowid` specially -- I wonder if that's worth changing? https://github.com/intel-ai/hdk/blob/main/omniscidb/QueryEngine/ExpressionRange.cpp#L547

Reply:
Looks like a bug we might hit someday. I guess the referenced handling should go after the next if-statement, not before.

Reply:
Hah, you're probably right. I'll try it in a separate branch.